src/main/java/com/groupesan/project/java/scrumsimulator/mainpackage/impl/Blocker.java

```diff
@@ -1,48 +1,62 @@
 package com.groupesan.project.java.scrumsimulator.mainpackage.impl;
 
-import lombok.Setter;
-
+import java.util.ArrayList;
+import java.util.List;
 import java.util.UUID;
 
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
 public class Blocker {
-    @Setter
     private UUID id;
-    @Setter
     private String name;
-    @Setter
     private String description;
     private boolean resolved;
 
+    // List to store linked user stories
+    private List<UserStory> linkedUserStories = new ArrayList<>();
+
     public Blocker(UUID blockerId, String name, String description) {
         this.id = blockerId;
         this.name = name;
         this.description = description;
         this.resolved = false;
     }
 
-    public UUID getId() {
-        return id;
-    }
-
-    public String getName() {
-        return name;
+    /**
+     * Adds a user story to the list of linked user stories if it’s not already linked.
+     * @param userStory the user story to link
+     */
+    public void addLinkedUserStory(UserStory userStory) {
+        if (!linkedUserStories.contains(userStory)) {
+            linkedUserStories.add(userStory);
+        }
     }
 
-    public String getDescription() {
-        return description;
-    }
-
-    public boolean isResolved() {
-        return resolved;
+    /**
+     * Removes a user story from the list of linked user stories.
+     * @param userStory the user story to unlink
+     */
+    public void removeLinkedUserStory(UserStory userStory) {
+        linkedUserStories.remove(userStory);
     }
 
+    /**
+     * Resolves the blocker.
+     */
     public void resolve() {
         this.resolved = true;
     }
 
-
     // Method to register the blocker in the BlockerStore
-//    public void doRegister() {
-//        BlockerStore.getInstance().addBlocker(this); // Register this blocker in the store
-//    }
+    public void doRegister() {
+        BlockerStore.getInstance().addBlocker(this);
+    }
```

Comment:
This method has been commented to avoid issues when storing blockers in simulation.json, pull the latest code

Reply:
I'll comment this part.